### ISSUE TITLE
[BUG] Since today it doesn't work.

### ISSUE THREAD

**Issue by GeorgeMJ23** [EltonChou/TwitterMediaHarvest#150]
Twitter must have made some changes to its code, so the Addon(in Firefox that I tried) it doesn't work. Even the download icon does not appear anymore.

**Comment by m1n1vv** [comment]
Yes. It's a similar problem in Edge. The update is only available for Chrome for now. But the download is not working.

**Comment by EltonChou** [comment]
> Twitter must have made some changes to its code, so the Addon(in Firefox that I tried) it doesn't work. Even the download icon does not appear anymore.

Firefox addon will be provided soon. (In release page)

**Comment by EltonChou** [comment]
> Yes. It's a similar problem in Edge. The update is only available for Chrome for now. But the download is not working.

Where is your ip location?

**Comment by m1n1vv** [comment]
> > Да. В Edge аналогичная проблема. Обновление пока доступно только для Chrome. Но загрузка не работает.
> 
> Где находится ваш IP-адрес?

Ukraine

**Comment by m1n1vv** [comment]
> > Yes. It's a similar problem in Edge. The update is only available for Chrome for now. But the download is not working.
> 
> Where is your ip location?

Is it possible to fix it?

**Comment by m1n1vv** [comment]
> > > > Yes. It's a similar problem in Edge. The update is only available for Chrome for now. But the download is not working.
> > > 
> > > 
> > > Where is your ip location?
> > 
> > 
> > Is it possible to fix it?
> 
> It will take sometime, but I think I can fix it.

I don't know how that's gonna help. Here is the message in the extension errors:

`TwitterApiError: Your login session might be expired, please refresh your page.(statusCode: 403)`
```
Context
sw.js
Stack trace
sw.js:2 (anonymous functions)
```

And I have another video download extension installed. It is working steadily now.

**Comment by adeteejay** [comment]
I just uninstalled and reinstalled and it's working again

**Comment by Nekodukin** [comment]
It doesn't work for me in Japan.
The FQDN for the PC version of the site has changed from twitter.com to x.com, and it appears that the extension no longer works.
It seems that the official X has announced the transition to x.com.

**Comment by Corleonis6422** [comment]
I also encountered the issue in Taiwan IP, reinstall the app doesn't resolve the problem.

**Comment by EltonChou** [comment]
The patch would be published tonight.

**Comment by alaa-ayyad** [comment]
Version 4.2.6 works Firefox, thanks!

One note: It started a brand new history repository, is it possible to continue using the existing history db?

**Comment by EltonChou** [comment]
> Version 4.2.6 works Firefox, thanks!
> 
> One note: It started a brand new history repository, is it possible to continue using the existing history db?

Did you migrate from `addons.mozilla.org` version?

**Comment by alaa-ayyad** [comment]
> > Version 4.2.6 works Firefox, thanks!
> > One note: It started a brand new history repository, is it possible to continue using the existing history db?
> 
> Did you migrate from `addons.mozilla.org` version?

Thanks for the reply. I just used the xpi file in the release as recommended in readme.

I downloaded the source and built it but didn't know were to place the generated files under /dist.

**Comment by EltonChou** [comment]
Since the id of addon was not changed, it should use same repository.
What if you install an older version? Is the history changed?

**Comment by alaa-ayyad** [comment]
Yes, the sqlite files are created in different storage folder for each xpi version.
Please see the screenshot below, I tried to move the sqlite files from 4.2 to 4.2.6 folder, the older history list appears but without thumbnails, and the id in the About section is missed up (all x's),

![explorer_pg5M6ccXpk](https://github.com/EltonChou/TwitterMediaHarvest/assets/12691011/b7c5378e-32a3-45d5-a990-f8e64dd6cdf3)

**Comment by EltonChou** [comment]
v4.2 is last AMO store version with extension id  `mediaharvest@addons.mozilla.org`
v4.2.6 is self-signed version with extension id `mediaharvest@mediaharvest.app`

Firefox will treat them as different addons.
The history is stored in indexed DB, if you know how to manipulate it you can migrate it manually.

**Comment by alaa-ayyad** [comment]
I see, I noticed the difference in the id name in preference files. I will try to do the migration manually. Thank you.

**Comment by alaa-ayyad** [comment]
I was able to migrate the db:

- Copied the .**sqlite**, **.sqlite-shm**, and .**sqlite-wal** files from C:`\Users\<user>\AppData\Roaming\Mozilla\Firefox\Profiles\<your_profile>\storage\default\moz-extension+++<OLD_extension_id>\idb`
to
`C:\Users\<user>\AppData\Roaming\Mozilla\Firefox\Profiles\<your_profile>\storage\default\moz-extension+++<NEW_extension_id>\idb`
 
- Open the copied file using any SQLite viewer tool

- Open table `database`

- Under `origin` column edit the cell and change id part to the NEW_extension_id value:
moz-extension://**NEW_extension_id**

Now you will be able to see the old history with working thumbnails and links.

**Comment by Gabarito007** [comment]
> The history is stored in indexed DB, if you know how to manipulate it you can migrate it manually.


Off topic:

How to increase the number of items on the History page?
Currently, it shows only 20 entries.

**Comment by EltonChou** [comment]
> 
> How to increase the number of items on the History page?
> 
> Currently, it shows only 20 entries per page.
> 
> 

I will add more options.
But it will take some time, I'm not good at UI.

**Comment by Gabarito007** [comment]
> I will add more options. But it will take some time, I'm not good at UI.

Great!
Many thanks.

**Comment by gpion** [comment]
Still not working for me on Edge, extension version 4.2.5 (chrome)

**Comment by lentilwallop** [comment]
when should we expect v4.2.6 to appear on the chrome web store?

**Comment by EltonChou** [comment]
> when should we expect v4.2.6 to appear on the chrome web store?

Based on past experience, the review will take 1 - 5 days.

**Comment by EltonChou** [comment]
> Strange. I can download media files of some posts, but some not.
> 
> (Firefox)

Please provdies those failed post.

**Comment by EltonChou** [comment]
> when should we expect v4.2.6 to appear on the chrome web store?

From now on.

**Comment by Tamerlanchiques** [comment]
For example, I downloaded video from [here](https://x.com/i/web/status/1792298634291093642) and tried to do this again but I lose download button.

As of 20 of May:

![image](https://github.com/EltonChou/TwitterMediaHarvest/assets/11006352/66f6610f-faa4-4a01-8ceb-f99c27f59cef)

As of now: 

![image](https://github.com/EltonChou/TwitterMediaHarvest/assets/11006352/d2f6e229-4398-4655-989e-6c8a21eb9bcd)

Firefox Mac 126.0 (64-bit)
Version 4.2.0 (firefox)

Yep, I noticed that I use old version but still strange that I downloaded something at 20 of May…

Long before I found your extention in Firefox addons site but could not find it now and can't download latest update. Why not host addon on Firefox Addons site?

**Comment by EltonChou** [comment]
> For example, I downloaded video from [here](https://x.com/i/web/status/1792298634291093642) and tried to do this again but I lose download button.
> 
> As of 20 of May:
> 
> ![image](https://private-user-images.githubusercontent.com/11006352/332584565-66f6610f-faa4-4a01-8ceb-f99c27f59cef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYzNDkwNDcsIm5iZiI6MTcxNjM0ODc0NywicGF0aCI6Ii8xMTAwNjM1Mi8zMzI1ODQ1NjUtNjZmNjYxMGYtZmFhNC00YTAxLThjZWItZjk5YzI3ZjU5Y2VmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIyVDAzMzIyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAwOGM5MzJlYjkxMDA1M2YwYThhMDAzMDBmYjg3NjUyNjUxZGE2MjI1YzBhN2U3OGE1N2U4ZjMzNzNiZGMxZGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.PFXD02ARyPUi8huJk4apmcfUxx6uadt_Ce6bIKag9xg)
> 
> As of now:
> 
> ![image](https://private-user-images.githubusercontent.com/11006352/332584622-d2f6e229-4398-4655-989e-6c8a21eb9bcd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYzNDkwNDcsIm5iZiI6MTcxNjM0ODc0NywicGF0aCI6Ii8xMTAwNjM1Mi8zMzI1ODQ2MjItZDJmNmUyMjktNDM5OC00NjU1LTk4OWUtNmM4YTIxZWI5YmNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIyVDAzMzIyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQyYTFiNzI5NzljNDExMDljNmQ0ZDJlZGRmMDNiYTBhNTg1NDI0NjVjMTZmMWUzZmRkODJhOGZiMmJlNDk4NDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.iUuA0Ta3BoL4RnNRSQadxG96xYWNZ7eN-RLktc0AXeE)
> 
> Firefox Mac 126.0 (64-bit) Version 4.2.0 (firefox)
> 
> Yep, I noticed that I use old version but still strange that I downloaded something at 20 of May…
> 
> Long before I found your extention in Firefox addons site but could not find it now and can't download latest update. Why not host addon on Firefox Addons site?

Latest version `.xpi` file is in release page.

Firefox AMO ref: #112

**Comment by EltonChou** [comment]
`v4.2.6` should fix this issue.
